### PR TITLE
feat: Introduce compatibility with native namespace packages

### PR DESCRIPTION
**Pull request by parthea**
Similar to https://github.com/googleapis/google-auth-library-python/pull/1205
Also see https://packaging.python.org/en/latest/guides/packaging-namespace-packages/#native-namespace-packages

### CODE REVIEW

File: testing/constraints-3.11.txt [hunk 1]
```diff
@@ -1,24 +1,13 @@
...
-#
-# Copyright 2020 Google LLC
+# Copyright 2024 Google LLC
 #
 # Licensed under the Apache License, Version 2.0 (the "License");
 # you may not use this file except in compliance with the License.
 # You may obtain a copy of the License at
 #
-#     https://www.apache.org/licenses/LICENSE-2.0
+#     http://www.apache.org/licenses/LICENSE-2.0
```
Comment (vchudnov-g) [inline testing/constraints-3.11.txt:7]: nit: both `http` and `https` work

File: tests/unit/test_packaging.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import subprocess
+import sys
+
+
+def test_namespace_package_compat(tmp_path):
```
Comment (vchudnov-g) [inline tests/unit/test_packaging.py:20]: Q: how does the testing infra know that `tmp` path is a `pathlib.Path`?

**Review by vchudnov-g — APPROVED** [review]

File: testing/constraints-3.11.txt (hunk 1 above)
Comment (parthea) [reply testing/constraints-3.11.txt:7]: `http` matches the license [here](https://www.apache.org/licenses/LICENSE-2.0#apply) which could be why we use that in templates such as https://github.com/googleapis/gapic-generator-python/blob/main/gapic/templates/_license.j2

File: tests/unit/test_packaging.py (hunk 2 above)
Comment (parthea) [reply tests/unit/test_packaging.py:22]: I added a comment in https://github.com/googleapis/python-audit-log/pull/117/commits/433522ce2796ed15c4239f8a1eaae74b774834a9 with more information about the `tmp_path` fixture of `pytest`

**Pull request merged**